app/src/main/java/com/sun/ev_dictionary/ui/splash/SplashActivity.kt

```diff
@@ -0,0 +1,80 @@
+package com.sun.ev_dictionary.ui.splash
+
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.sun.ev_dictionary.R
+import com.sun.ev_dictionary.base.BaseActivity
+import com.sun.ev_dictionary.data.source.local.EnglishWordDatabase
+import com.sun.ev_dictionary.data.source.local.EnglishWordsLocalDataSource
+import com.sun.ev_dictionary.data.source.repository.EnglishWordsRepository
+import com.sun.ev_dictionary.databinding.ActivitySplashBinding
+import com.sun.ev_dictionary.ui.home.HomeActivity
+import com.sun.ev_dictionary.utils.Constants
+import com.sun.ev_dictionary.utils.Constants.PREF_ENGLISH_WORDS
+import com.sun.ev_dictionary.utils.SharedPreference
+
+class SplashActivity : BaseActivity<ActivitySplashBinding>() {
+    private lateinit var viewModel: SplashViewModel
+    private lateinit var viewModelFactory: SplashViewModelFactory
+    private lateinit var sharedPreference: SharedPreference
+    override fun getLayoutResource(): Int = R.layout.activity_splash
+
+    override fun initData() {
+        sharedPreference = SharedPreference(this)
+        val bufferedReadable = applicationContext.assets
+            .open(Constants.FILE_NAME_EV)
+            .bufferedReader()
+        viewModelFactory = SplashViewModelFactory(
+            EnglishWordsRepository.getInstance(
+                EnglishWordsLocalDataSource.getInstance(
+                    bufferedReadable,
+                    EnglishWordDatabase.getInstance(this).englishWordDao
+                )
+            )
+        )
+        viewModel = ViewModelProviders.of(this, viewModelFactory)
```

Comment:
viewmodel em nên add thêm lifycleOwner cho nó nữa.

Reply:
chỗ này e rõ lắm ạ. E tưởng viết ở trong BaseActivity là như này là đc rồi chứ ạ:
override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, getLayoutResource())
        binding.lifecycleOwner = this
        initData()
    }